src/EnergyPlus/HVACControllers.cc

```diff
@@ -1046,42 +974,41 @@ void InitController(EnergyPlusData &state, int const ControlNum, bool &IsConverg
+        } else if (thisController.MinVolFlowActuated >= thisController.MaxVolFlowActuated) {
             ShowFatalError(state,
                            format("{}: Controller:WaterCoil=\"{}\", Minimum control flow is > or = Maximum control flow.",
                                   RoutineName,
-                                  ControllerProps(ControlNum).ControllerName));
+                                  thisController.ControllerName));
         }
 
         // Setup root finder after sizing calculation
-        switch (ControllerProps(ControlNum).Action) {
+        switch (thisController.Action) {
         case ControllerAction::NormalAction: {
-            SetupRootFinder(state,
-                            RootFinders(ControlNum),
-                            DataRootFinder::Slope::Increasing,
-                            DataRootFinder::RootFinderMethod::Brent,
-                            DataPrecisionGlobals::constant_zero,
-                            1.0e-6,
-                            ControllerProps(ControlNum).Offset); // Slope type | Method type | TolX: no relative tolerance for X variables |
-                                                                 // ATolX: absolute tolerance for X variables | ATolY: absolute tolerance for
-                                                                 // Y variables
+            RootFinder::SetupRootFinder(state,
+                                        state.dataHVACControllers->RootFinders(ControlNum),
+                                        DataRootFinder::Slope::Increasing,
+                                        DataRootFinder::RootFinderMethod::Brent,
+                                        DataPrecisionGlobals::constant_zero,
+                                        1.0e-6,
+                                        thisController.Offset); // Slope type | Method type | TolX: no relative tolerance for X variables |
+                                                                // ATolX: absolute tolerance for X variables | ATolY: absolute tolerance for
+                                                                // Y variables
 
         } break;
         case ControllerAction::Reverse: {
-            SetupRootFinder(state,
-                            RootFinders(ControlNum),
-                            DataRootFinder::Slope::Decreasing,
-                            DataRootFinder::RootFinderMethod::Brent,
-                            DataPrecisionGlobals::constant_zero,
-                            1.0e-6,
-                            ControllerProps(ControlNum).Offset); // Slope type | Method type | TolX: no relative tolerance for X variables |
-                                                                 // ATolX: absolute tolerance for X variables | ATolY: absolute tolerance for
-                                                                 // Y variables
+            RootFinder::SetupRootFinder(state,
```

Comment:
Do we still need `RootFinder`? Isn't it just an API for implementing `SolveRoot`?